api/handlers.go

```diff
@@ -70,6 +70,165 @@ func CollectionHandler(c *gin.Context) {
 	concludeRequest(c, resp, err)
 }
 
+func LessonOverviewHandler(c *gin.Context) {
+	var r LessonOverviewRequest
+	if c.Bind(&r) != nil {
+		return
+	}
+	db := c.MustGet("MDB_DB").(*sql.DB)
+
+	//append collection filters
+	cMods := []qm.QueryMod{SECURE_PUBLISHED_MOD}
```

Comment:
For item #2: for now there are not seems we have reason to refactor that part because not that much sharable code. I believe after we add filters - we could extract some parts